src/rgw/driver/rados/rgw_log_backing.h

```diff
@@ -262,12 +262,24 @@ class LazyFIFO {
 
   int lazy_init(const DoutPrefixProvider *dpp, optional_yield y) {
     std::unique_lock l(m);
-    if (fifo) return 0;
-    auto r = rgw::cls::fifo::FIFO::create(dpp, ioctx, oid, &fifo, y);
-    if (r) {
-      fifo.reset();
+    if (fifo) {
+      return 0;
+    } else {
+      l.unlock();
+      // FIFO supports multiple clients by design, so it's safe to
+      // race to create them.
+      std::unique_ptr<rgw::cls::fifo::FIFO> fifo_tmp;
+      auto r = rgw::cls::fifo::FIFO::create(dpp, ioctx, oid, &fifo, y);
```

Comment:
since the intent was to initialize a separate `fifo_tmp` to prevent multiple callers from racing on member variable `fifo`, this should have passed `&fifo_tmp` instead. this seems to be the cause of https://tracker.ceph.com/issues/66880

cc @smanjara

Reply:
see https://github.com/ceph/ceph/pull/58491